<launch>
 <arg name="x_0" default="10"/>
 <arg name="y_0" default="9"/>
 <arg name="x_1" default="15"/>
 <arg name="y_1" default="13"/>

 <node name="rviz" pkg="rviz" type="rviz" args="-d $(find map_transform)/launch/pddl_twoRobots.rviz" />

 <node pkg="map_server" type="map_server" name="map_server" args="$(find map_transform)/maps/maze.png 0.1" respawn="false" > 
    <param name="frame_id" value="/map" />
 </node>

	
 <group ns="robot_0">

    <node pkg="map_transform" type="visNC" name="visibility" output="screen" args="$(find map_transform)/robots/robot2.png $(find map_transform)/robots/robot2.png">    
       <param name="sdefl" value="60" />
       <param name="rinfl" value="80" />
       <param name="rx" value="50" />
       <param name="ry" value="50" />
       <param name="rt" value="0" />
       <param name="sx" value="50" />
       <param name="sy" value="50" />
       <param name="st" value="0" />
       <param name="dx" value="50" />
       <param name="dy" value="50" />
       <param name="angle_res" value="64" />
       <param name="sens_res" value="1" />
       <param name="debug" value="false" />
       <param name="show" value="false" />
       <param name="ground_truth" value="false" />
       <param name="act" value="true" />
       <param name="debug_angle" value="0" />
       <param name="tf_prefix" value ="/robot_0" /> 
       <remap from="/robot_0/visibility/map" to="/map"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="fake_localize_0" args="$(arg x_0) $(arg y_0) 0 0 0 0 /map /robot_0/base_link 10"/>

 </group>

 <group ns="robot_1">

    <node pkg="map_transform" type="visNC" name="visibility" output="screen" args="$(find map_transform)/robots/robot2.png $(find map_transform)/robots/robot2.png">    
       <param name="sdefl" value="100" />
       <param name="rinfl" value="120" />
       <param name="rx" value="50" />
       <param name="ry" value="50" />
       <param name="rt" value="0" />
       <param name="sx" value="50" />
       <param name="sy" value="50" />
       <param name="st" value="0" />
       <param name="dx" value="50" />
       <param name="dy" value="50" />
       <param name="angle_res" value="64" />
       <param name="sens_res" value="1" />
       <param name="debug" value="false" />
       <param name="show" value="false" />
       <param name="ground_truth" value="false" />
       <param name="act" value="true" />
       <param name="debug_angle" value="0" />
       <param name="tf_prefix" value ="/robot_1" /> 
       <remap from="/robot_1/visibility/map" to="/map"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="fake_localize_1" args="$(arg x_1) $(arg y_1) 0 0 0 0 /map /robot_1/base_link 10"/>

 </group>

 <node name="pddl_generator" pkg="map_transform" type="pddl-gen-NC" cwd="node" output="screen">
       	<param name="jump" value="15" />
       	<remap from="/robot_0/c_map" to="/robot_0/visibility/c_map"/>
       	<remap from="/robot_1/c_map" to="/robot_1/visibility/c_map"/>
       	<remap from="/robot_0/e_map" to="/robot_0/visibility/e_map"/>
       	<remap from="/robot_1/e_map" to="/robot_1/visibility/e_map"/>
       	<remap from="/robot_0/a_map" to="/robot_0/visibility/a_map"/>
       	<remap from="/robot_1/a_map" to="/robot_1/visibility/a_map"/>
       	<remap from="/robot_0/r_map" to="/robot_0/visibility/r_map"/>
       	<remap from="/robot_1/r_map" to="/robot_1/visibility/r_map"/>
       	<remap from="/robot_0/pc_map" to="/robot_0/visibility/pc_map"/>
       	<remap from="/robot_1/pc_map" to="/robot_1/visibility/pc_map"/>
       	<remap from="/robot_0/pe_map" to="/robot_0/visibility/pe_map"/>
       	<remap from="/robot_1/pe_map" to="/robot_1/visibility/pe_map"/>
       	<remap from="/robot_0/pa_map" to="/robot_0/visibility/pa_map"/>
       	<remap from="/robot_1/pa_map" to="/robot_1/visibility/pa_map"/>
       	<remap from="/robot_0/pr_map" to="/robot_0/visibility/pr_map"/>
       	<remap from="/robot_1/pr_map" to="/robot_1/visibility/pr_map"/>
       	<remap from="/robot_0/act_map" to="/robot_0/visibility/act_dist"/>
       	<remap from="/robot_1/act_map" to="/robot_1/visibility/act_dist"/>
       	<remap from="/robot_0/c_multi" to="/robot_0/visibility/c_multi_level"/>
       	<remap from="/robot_1/c_multi" to="/robot_1/visibility/c_multi_level"/>
       	<remap from="/robot_0/e_multi" to="/robot_0/visibility/e_multi_level"/>
       	<remap from="/robot_1/e_multi" to="/robot_1/visibility/e_multi_level"/>
       	<remap from="/robot_0/a_multi" to="/robot_0/visibility/a_multi_level"/>
       	<remap from="/robot_1/a_multi" to="/robot_1/visibility/a_multi_level"/>
       	<remap from="/robot_0/r_multi" to="/robot_0/visibility/r_multi_level"/>
       	<remap from="/robot_1/r_multi" to="/robot_1/visibility/r_multi_level"/>
       	<remap from="/robot_0/rob_struct" to="/robot_0/visibility/rob_struct"/>
       	<remap from="/robot_1/rob_struct" to="/robot_1/visibility/rob_struct"/>
       	<remap from="/robot_0/act_struct" to="/robot_0/visibility/act_struct"/>
       	<remap from="/robot_1/act_struct" to="/robot_1/visibility/act_struct"/>
       	<remap from="/robot_0/rob_center" to="/robot_0/visibility/rob_center"/>
       	<remap from="/robot_1/rob_center" to="/robot_1/visibility/rob_center"/>
 </node>

</launch>
